vault/seal_autoseal.go

```diff
@@ -87,8 +87,9 @@ func (d *autoSeal) Finalize(ctx context.Context) error {
 	return d.Access.Finalize(ctx)
 }
 
-func (d *autoSeal) BarrierType() string {
-	return d.Type()
+func (d *autoSeal) BarrierType() wrapping.WrapperType {
+	typ, _ := d.Type(context.Background())
```

Comment:
Might be nice to change the wrapper.Type api to not return an error? 

A bit worrisome that we are ignoring the error returned possibly in the future, none of the implementations I saw will ever return an error at the moment.

Reply:
Yeah, I didn't love this either, but I don't think we can change wrapping.Type.  Wonder if I can at least log an error here, let me look.

Reply:
Addressed by latest commit, thanks Scott.